bin-src/types.ts

```diff
@@ -0,0 +1,276 @@
+  only?: string;
+  onlyChanged?: string;
+  patchBuild?: string;
+  preserveMissing?: boolean;
+  skip?: string;
+  storybookBaseDir?: string;
+  storybookConfigDir?: string;
+  untraced?: string[];
+  zip?: boolean;
+
+  // Debug options
+  debug?: boolean;
+  diagnostics?: boolean;
+  dryRun?: boolean;
+  forceRebuild?: string;
+  junitReport?: string;
+  list?: boolean;
+  interactive?: boolean;
+  traceChanged?: string;
+
+  // Deprecated options (for JSDOM and tunneled builds)
+  allowConsoleErrors?: boolean;
+  appCode?: string[];
+  doNotStart?: boolean;
+  exec?: string;
+  scriptName?: string;
+  storybookPort?: string;
+  storybookUrl?: string;
+  storybookHttps?: boolean;
+  storybookCert?: string;
+  storybookKey?: string;
+  storybookCa?: string;
+}
+
+export interface Options {
+  projectToken: string;
+
+  only: Flags['only'];
+  onlyChanged: true | string;
+  untraced: Flags['untraced'];
+  externals: Flags['externals'];
+  traceChanged: true | string;
+  list: Flags['list'];
+  fromCI: boolean;
+  skip: true | string;
+  dryRun: Flags['dryRun'];
+  forceRebuild: true | string;
+  verbose: boolean;
+  interactive: boolean;
+  junitReport: true | string;
+  zip: Flags['zip'];
+
+  autoAcceptChanges: true | string;
+  exitZeroOnChanges: true | string;
+  exitOnceUploaded: true | string;
+  ignoreLastBuildOnBranch: Flags['ignoreLastBuildOnBranch'];
+  preserveMissingSpecs: boolean;
+  originalArgv: string[];
+
+  buildScriptName: Flags['buildScriptName'];
+  outputDir: string;
+  allowConsoleErrors: Flags['allowConsoleErrors'];
+  scriptName: string;
+  exec: Flags['exec'];
+  noStart: Flags['doNotStart'];
+  https: {
+    cert: Flags['storybookCert'];
+    key: Flags['storybookKey'];
+    ca: Flags['storybookCa'];
+  };
+  url?: string;
+  port: Flags['storybookPort'];
+  storybookBuildDir: string;
+  storybookBaseDir: Flags['storybookBaseDir'];
+  storybookConfigDir: Flags['storybookConfigDir'];
+  storybookUrl: Flags['storybookUrl'];
+  createTunnel: boolean;
+  useTunnel?: boolean;
+
+  ownerName: string;
+  branchName: string;
+  patchHeadRef: string;
+  patchBaseRef: string;
+}
+
+export interface Context {
+  env: Env;
+  log: Logger;
+  pkg: {
+    name: string;
+    version: string;
+    description: string;
+    bugs: { url: string; email: string };
+    docs: string;
+  };
+  sessionId: string;
+  packageJson: { [key: string]: any };
+  packagePath: string;
+  help: any;
+  argv: string[];
+  flags: Flags;
+  options: Options;
+  title: string;
+  skip?: boolean;
+  skipSnapshots?: boolean;
+  now?: number;
+  startedAt?: number;
+  exitCode: number;
+  exitCodeKey: string;
+  userError?: boolean;
+  runtimeErrors?: Error[];
+  runtimeWarnings?: Error[];
+  environment?: string;
+  reportPath?: string;
+  stopApp?: () => void;
+  closeTunnel?: () => void;
+  isPublishOnly?: boolean;
+  isOnboarding?: boolean;
+
+  http: {
+    fetch: (url: string, options?: Record<string, any>, opts?: any) => Promise<Response>;
```

Comment:
Is this using node-fetch or something? I would assume so.. might just expose the real types for that.

Reply:
Done 👍